tests/test_ext_marshmallow.py

```diff
@@ -451,11 +456,51 @@ def pet_view():
         p = spec._paths['/pet']
         assert 'get' in p
         op = p['get']
+        assert 'parameters' in op
+        len(op['parameters']) == 1
         assert 'responses' in op
-        assert op['responses'][200]['schema'] == {
+        resolved_schema = {
             'type': 'array',
             'items': {'$ref': '#/definitions/Pet'}
         }
+        assert op['parameters'][0]['schema'] == resolved_schema
+        assert op['responses'][200]['schema'] == resolved_schema
+
+    def test_schema_array_in_docstring_uses_ref_if_available_v3(self, spec_3):
+        def pet_view():
```

Comment:
I think this tests passes because `spec3.definition('Pet', schema=PetSchema)` is done in an earlier test.

Reply:
Sorry, I take that back. It passes even if ran alone.

It differs from the v2 test in that the v2 checks for the reference `{'$ref': '#/definitions/Pet'}` explicitly. AFAIU, this test does not check that the reference is used. Or it does, if the ref exists, but we're not sure about that.

I rebased the "classes Plugin" branch (https://github.com/marshmallow-code/apispec/issues/207) on this and I think it is fixed there.

Reply:
@lafrech let me know if you'd like me to change something here. I can adjust the test to check the reference with the following change, 

```python
        spec_3.definition('Pet', schema=PetSchema)
        spec_3.add_path(path='/pet', view=pet_view)
        ...
        resolved_schema = {
            'type': 'array',
            'items': {'$ref': '#/components/schemas/Pet'},
        }
        ...
```

Reply:
Yes, exactly. Like the v2 test above. This is what I did in https://github.com/marshmallow-code/apispec/tree/dev_207_plugin_classes.